### ISSUE TITLE
Option to Export to fixed file in a specific Worksheet?

### ISSUE THREAD

**Issue by BionicThomas** [ma4nn/moneymoney-sum-by-bank#1]
Hi Christoph!

Could you point me towards a solution where the data is not written to a new file but an existing file?
I have a self-made Budget and Forecast for my Financials in Excel and it would be great to get the actuals automatically.
Currently I manually type them into my Excel file from MoneyMoney.
Your script seems to be perfect with just this little change.

Many thanks!

**Comment by BionicThomas** [comment]
OK, changed your script to have a fixed file and save and close this one.
As I added a different bankIdentifier per account I basically get an export by account out of your script.
Works great so far.
Thanks!

```
-- Open Microsoft Excel application, insert bank sums and format and sort accordingly
on OpenExcelWithData(bankBalances)
	set file_Name to "/Users/<user>/Documents/MoneyMoneyExport.xlsx"
	tell application "Microsoft Excel"
		activate
		open file_Name
		-- make new workbook
		set x to 1
		
		repeat with balanceData in bankBalances
			set balance to balance of balanceData
			set bank to bankIdentifier of balanceData
			
			set value of cell x of column 1 to balance
			set value of cell x of column 2 to bank
			set x to (x + 1)
		end repeat
		
		set number format of range "$A1:$A100" to "#,##0.00 €"
		-- 	sort range "A1:B10" key1 range "A1" order1 sort descending
		save yes
		close active workbook
	end tell
end OpenExcelWithData

```

**Comment by ma4nn** [comment]
Hi @BionicThomas,

cool, thanks very much for your contribution!

I have created a new Pull Requests that contains your changes by introducing some configuration options at the top of the AppleScript. 
Would be really cool, if you could have a look at the linked PR and tell me if this is appropriate for your use case.

Best regards,
Christoph

**Comment by BionicThomas** [comment]
Hi Christoph!

Now that I started to trust the script I tested as well to use a sheet within my Budget Excel file instead.

Added another parameter
`set exportWorkSheet to "MoneyMoneyExport"`

Changed the Method declaration to accept an additional parameter

`OpenExcelWithData(bankBalances, fileName, targetWorkSheet, startRowIndex, startColumnIndex, isSortDescending, isCloseExcel)`

and added another if statement with code to select that worksheet.

`if (fileName is missing value) then
			make new workbook
		else
			open fileName
			if exists sheet targetWorkSheet then
				select sheet targetWorkSheet
			end if
		end if`

the call of the method had to be changed, too as you will already know.

`OpenExcelWithData(bankBalances, exportFileName, exportWorkSheet, startRowIndex, startColumnIndex, isSortDescending, isCloseExcel)`

Worked fine!

btw. I only found your script by accident after days of search and trying myself. Might be worth getting this listed as a plugin on the authors web site to let others benefit from it, too.
